lms/services/digest.py

```diff
@@ -56,11 +65,20 @@ def send_instructor_email_digests(
                 # We don't have an email address for this user.
                 continue
 
+            if self._email_unsubscribe_service.is_unsubscribed(
+                to_email, EmailUnsubscribeService.Tag.INSTRUCTOR_DIGEST
+            ):
+                # `to_email` unsubscribed from this type of email
+                continue
```

Comment:
I don't think we want to do this here as it'll mean an extra DB query for each user in the batch, whereas `DigestService` currently does a fixed number of DB queries regardless of the batch size.

Also by the time we get to this point (we're about to send the email to the user) we've already done a lot of unnecessary work: fetching this user's annotations from h, making DB queries and generating this user's digest.

Instead I think we probably want to filter unsubscribed users in [the periodic task that finds all the users to be emailed](https://github.com/hypothesis/lms/blob/415351f1460e07b8a9da891b0c2eceb1c6e60081/lms/tasks/email_digests.py#L14-L65) in the first place. I could see this working in either of two ways:

1. You could include the new `EmailUnsubscribe` model in the DB query that the task does, so that it filters out the `h_userid`'s that have an unsubscribe for their email
2. Or you could do two separate queries:
 
   1. The existing query to find all the `h_userid`'s, unchanged
   2. A second query to find out which of those `h_userid`'s have unsubscribes
   
   Then filter the `h_userid`'s in Python before doing the batching and before calling `send_instructor_email_digests.apply_async()`: `h_userids = [h_userid for h_userid in h_userids if h_userid not in unsubscribed_h_usierds]`